src/wrapper/objects/auto_primitive_array.rs

```diff
@@ -46,7 +46,8 @@ impl<'a, 'b> AutoPrimitiveArray<'a, 'b> {
         self.ptr.as_ptr()
     }
 
-    fn commit_primitive_array_critical(&mut self) -> Result<()> {
+    /// Commits the changes to the array, if it is a copy
+    pub fn commit(&mut self) -> Result<()> {
         jni_void_call!(
```

Comment:
Wouldn't similar changes to deduplicate the methods make sense here?

Reply:
IMO, no. Exacly because of the naming conventions. We shouldn't carry or reflect a mistake or poor design decision in the underlying API to our public API. This is opinionated, I know; and that's my opinion. :-)

Reply:
Ah, you're thinking about doing the same as before but for primitive array... Sure! good catch, I thought I already did that.